docs/sdk/instructlab-sdk.md

```diff
@@ -0,0 +1,109 @@
+# InstructLab SDK
+
+## Motivation
+
+Today, the only way to "drive" the InstructLab opinionated workflow is via the `ilab` CLI. While this process provides a succinct way for everyday users to initialize a config, generate synthetic data, train a model, and evaluate it: the guardrails are quite limiting both in what a user can do and what the development team can add as features exposed directly to the user over time.
+
+Additionally, current consumers of InstructLab are finding themselves importing different library private and public APIs in combination with CLI functionality to achieve the workflows they want. While these more advanced usage patterns are not for everyone, providing ways to run bespoke and piecemeal workflows in a standardized and safe way for the community is a necessity.
```

Comment:
What's the mechanism to collect these real use patterns from external consumers, to make sure their needs are properly served?

Reply:
good question. Most of our usage information is coming from internal use cases combined with reports of what people are asking for colloquially. We have no request gathering system as of yet

Reply:
Understood. I'd advise then to at least have several interview sessions with potential consumers, collecting requests / feedback / pain points.